chall07/gbudau.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+)
+
+
+func requestServer() {
+	/* First GET request to get the required formated string */
+	resp, err := http.Get("https://chall07.42madrid.com/")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Response status:", resp.Status)
+	defer resp.Body.Close()
+
+	/* Read and print response */
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("%s\n", string(body))
+
+	/* Use regex to find the numbers in the string and check to be more than 3 numbers */
+	/* Expecting the first 4 numbers to be in correct order and valid */
+	re := regexp.MustCompile(`\d+`)
+	set := re.FindAllString(string(body), -1)
+	if len(set) < 4 {
+		panic(err)
+	}
+
+	/* Convert second, third and forth digits to int and create an "answer" string*/
+	/* Expecting all colors to be between 0-255 */
+	r, _ := strconv.Atoi(set[1])
+	g, _ := strconv.Atoi(set[2])
+	b, _ := strconv.Atoi(set[3])
```

Comment:
A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, r, g, b int
	body := send_get_from_server("https://chall07.42madrid.com/")
	if _, err := fmt.Sscanf(string(body), "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &r, &g, &b); err != nil {
		panic(err)
	}

Reply:
:smile:

That looks so much better and simpler, thank you!